src/utils/validations/component/validations/form-with-validations.hoc.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import ValidationsContext from './validations.context';
+
+const withValidations = (WrappedComponent) => {
+  return class extends React.Component {
+    state = {
+      errorCount: 0,
+      warningCount: 0,
+      infoCount: 0
+    }
+
+    inputs = {};
+
+    validationTypes = ['validations', 'warnings', 'info'];
+
+    addInput = (name, validate) => {
+      this.inputs[name] = validate;
+    }
+
+    removeInput = (name) => {
+      delete this.inputs[name];
+    }
+
+    adjustCount = (type, adjustment) => {
+      switch (type) {
+        case 'error':
+          this.setState((prev) => { return { errorCount: prev.errorCount + adjustment }; });
```

Comment:
Just wondering about the reason for the function. I would probably write this as:

`this.setState({errorCount: this.state.errorCount + adjustment };`

Reply:
The linter complains when you don't use a callback when referencing previous state, which will make the build fail